packages/hydrogen/src/cart/cartGetIdDefault.ts

```diff
@@ -1,7 +1,12 @@
 import {parse} from 'worktop/cookie';
+import type {CrossRuntimeRequest} from '../utils/request';
 
-export const cartGetIdDefault = (requestHeaders: Headers) => {
-  const cookies = parse(requestHeaders.get('Cookie') || '');
+export const cartGetIdDefault = (
+  requestHeaders: Headers | CrossRuntimeRequest['headers'],
+) => {
+  const cookies = parse(
+    (requestHeaders.get ? requestHeaders.get('Cookie') : undefined) || '',
```

Comment:
You can use `getHeader` from `utils/request.ts`, it understands `CrossRuntimeRequest.headers` already.
We might need to change the type here to be `requestHeaders: CrossRuntimeRequest['headers']` though 🤔

Reply:
The idea is that `CrossRuntimeRequest['headers']` already contains both types. It basically requests only the minimum type that work in both runtimes. Am I wrong? 🤔

Reply:
I think you would just need to pass is in an object, like `getHeader({headers}, key)`, because the rest of the keys in CrossRuntimeRequest are optional. That said, `getHeaderValue` is OK as well.